include/swift/Serialization/SerializedModuleLoader.h

```diff
@@ -62,11 +62,17 @@ struct SerializedModuleBaseName {
   /// Gets the filename with a particular extension appended to it.
   std::string getName(file_types::ID fileTy) const;
 
-  /// If the interface with \p baseName exists, returns its path (which may be
-  /// the private interface if there is one). Return an empty optional
-  /// otherwise.
+  /// If the interface with \p baseName exists, returns its path (which may be the
+  /// package interface if applicable (in the same package as the main module) or
+  /// private interface if there is one, else public). Return an empty optional otherwise.
   llvm::Optional<std::string>
-  findInterfacePath(llvm::vfs::FileSystem &fs) const;
+  findInterfacePath(llvm::vfs::FileSystem &fs, ASTContext &ctx) const;
+  
+  /// Returns the .package.swiftinterface path if its package-name also applies to
+  /// the the importing module. Returns an empty optional otherwise.
+  llvm::Optional<std::string>
+  getPackageInterfacePathIfInSamePackage(llvm::vfs::FileSystem &fs,
```

Comment:
Nit: It feels a bit overkill to pass in the whole Context here. I think we just need an optional package name?

Reply:
It checks if Ctx.LangOpts.EnablePackageInterfaceLoad as well. I can pass this boolean and package name as args though if preferred.